### PR TITLE
Refine content and appearance of the user account screen

### PR DESCRIPTION
**Pull request by janfaracik**
This pull request updates the content and minor styling of the user account screen to make the content a little cleaner and inline. This is part of an ongoing effort to simplify Jenkins' many forms.

**Changes**
* Reduced usages of the❓button in the form, preferably if help guidance is short, snappy and useful it should be displayed inline
* Refined content for 'Full name', 'Description', 'Console URL Provider' and 'User Defined Time Zone'
* `UserPropertyDescriptor` now has a `getDescription()` method that can be overridden - these descriptions will show on pages where supported
* Wrapped the form in the `jenkins-form` class to fix sizing
* Grouped the first two controls (name and description) in a section so that they display with a horizontal divider below

---

[View comparison](https://janfaracik.github.io/comparison?before=https://github.com/user-attachments/assets/40fa4474-379b-4867-b369-efacdca36d00&after=https://github.com/user-attachments/assets/1df0c585-2451-4a3d-bc48-989742f57296&github=https://github.com/jenkinsci/jenkins/pull/9521)

---

**Before**

354807802-40fa4474-379b-4867-b369-efacdca36d00

---

**After**

354807736-1df0c585-2451-4a3d-bc48-989742f57296

### Testing done

* User config page works as before

### Proposed changelog entries

- Refine content and appearance of the user account screen.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by timja** [comment]
FYI something is wrong with your screenshots maybe transparency? in dark mode they are really messed up, and even in light mode if I open in a new tab they don't render right for me

File: core/src/main/resources/jenkins/console/Messages.properties
Comment (timja) [inline core/src/main/resources/jenkins/console/Messages.properties:2]: maybe? tools doesn't sound right

```suggestion
consoleUrlProviderDisplayName.Description=Select which view you want to use for build console links. Views listed higher will be prioritized over those listed lower. The views you choose here will be used rather than those set by administrators.
```

**Comment by janfaracik** [comment]
> FYI something is wrong with your screenshots maybe transparency? in dark mode they are really messed up, and even in light mode if I open in a new tab they don't render right for me

Should be fixed now, thanks!

**Review by daniel-beck — DISMISSED** [review]
It is unusual for a descriptor to always have a single `f:section`, this might be the only form where that's actually the case. Given limited applicability (a single `Descriptor` here?) it seems unnecessary to extend `Descriptor` for this use case. Please demonstrate this is useful well beyond the descriptors and forms shown in this PR.

**Comment by janfaracik** [comment]
> It is unusual for a descriptor to always have a single `f:section`, this might be the only form where that's actually the case. Given limited applicability (a single `Descriptor` here?) it seems unnecessary to extend `Descriptor` for this use case. Please demonstrate this is useful well beyond the descriptors and forms shown in this PR.

That's completely fair, I've moved it to UserPropertyDescriptor

**Review by timja — APPROVED** [review]
LGTM, tested and all appears sensible.

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
